test/tests/dgkernels/advection_diffusion_mixed_bcs_test_resid_jac/dg_advection_diffusion_test.i

```diff
@@ -0,0 +1,105 @@
+[Mesh]
+  type = GeneratedMesh
+  nx = 2
+  dim = 1
+[]
+
+[Kernels]
+  [./source]
+    type = UserForcingFunction
+    variable = u
+    function = 'forcing_func'
+  [../]
+  [./convection]
+    type = ConservativeAdvection
```

Comment:
I have added it into the framework with my second commit. I also think it should be in the framework. I use it frequently in my own applications for building test problems and even use it occasionally for real simulations.